File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int movi=0, swapi=0;
		while(movi < nums.size()) {
			if (nums[movi] != 0) {
			    swap(nums, swapi, movi);
			    swapi++; movi++;
			} else movi++;
		}
			
		while(swapi<nums.size()) nums[swapi++] = 0;
    }

private:
    void swap(vector<int>& arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public static void swap(int[] nums, int i , int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
    public void moveZeroes(int[] nums) {
        int p = 0 ;
        for(int sp = 0 ; sp < nums.length; sp++){
            if(nums[sp] != 0){
                swap(nums, p , sp);
                p++;
            }
        }
    }
}